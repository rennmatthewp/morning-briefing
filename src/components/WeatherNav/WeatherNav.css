.nav {
  height: 25px;
  max-width: 100%;
  padding: 5px 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.nav-link {
  background-color: #f2f2f2;
  border-radius: 5px; 
  width: fit-content;
  height: 20px;
  padding: 5px 5px 0;
  margin: 5px auto;
}

.weather-nav {
  width: 125px;
}

.active {
  background-color: #d1d1d1;
  font-weight: 900;
  border: solid .5px #000;
}