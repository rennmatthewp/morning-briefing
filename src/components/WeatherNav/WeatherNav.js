import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './WeatherNav.css';

export const WeatherNav = ({ selectCategory, section }) => {
  return (
    <div className="weather-nav nav">
      <NavLink
        to={`/${section}/hourly`}
        name="hourly"
        onClick={selectCategory}
        className="nav-link">
        Hourly
      </NavLink>
      <NavLink
        to={`/${section}/daily`}
        name="forecast10day"
        onClick={selectCategory}
        className="nav-link">
        Daily
      </NavLink>
    </div>
  );
};

WeatherNav.propTypes = {
  selectCategory: PropTypes.func,
  section: PropTypes.string
};
