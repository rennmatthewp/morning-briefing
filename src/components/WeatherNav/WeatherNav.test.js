import React from 'react';
import { shallow } from 'enzyme';
import { WeatherNav } from './WeatherNav';

describe('WeatherNav', () => {
  let renderedWeatherNav;
  let mockSelectCategory;

  beforeEach(() => {
    mockSelectCategory = jest.fn();
    renderedWeatherNav = shallow(
      <WeatherNav selectCategory={mockSelectCategory} />
    );
  });

  it('should match the snapshot', () => {
    expect(renderedWeatherNav).toMatchSnapshot();
  });

  it('should call selectSection onClick of a NavLink', () => {
    renderedWeatherNav
      .find('NavLink')
      .first()
      .simulate('click');
    expect(mockSelectCategory).toHaveBeenCalled();
  });
});
