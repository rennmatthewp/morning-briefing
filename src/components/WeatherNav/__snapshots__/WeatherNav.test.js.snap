// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherNav should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="nav-link selected"
          ariaCurrent="true"
          className="nav-link"
          name="hourly"
          onClick={[Function]}
          to="/hourly"
>
          Hourly
</NavLink>,
        <NavLink
          activeClassName="nav-link selected"
          ariaCurrent="true"
          className="nav-link"
          name="daily"
          onClick={[Function]}
          to="/daily"
>
          Daily
</NavLink>,
      ],
      "className": "weather-nav nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "nav-link selected",
          "ariaCurrent": "true",
          "children": "Hourly",
          "className": "nav-link",
          "name": "hourly",
          "onClick": [Function],
          "to": "/hourly",
        },
        "ref": null,
        "rendered": "Hourly",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "nav-link selected",
          "ariaCurrent": "true",
          "children": "Daily",
          "className": "nav-link",
          "name": "daily",
          "onClick": [Function],
          "to": "/daily",
        },
        "ref": null,
        "rendered": "Daily",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="nav-link selected"
            ariaCurrent="true"
            className="nav-link"
            name="hourly"
            onClick={[Function]}
            to="/hourly"
>
            Hourly
</NavLink>,
          <NavLink
            activeClassName="nav-link selected"
            ariaCurrent="true"
            className="nav-link"
            name="daily"
            onClick={[Function]}
            to="/daily"
>
            Daily
</NavLink>,
        ],
        "className": "weather-nav nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "nav-link selected",
            "ariaCurrent": "true",
            "children": "Hourly",
            "className": "nav-link",
            "name": "hourly",
            "onClick": [Function],
            "to": "/hourly",
          },
          "ref": null,
          "rendered": "Hourly",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "nav-link selected",
            "ariaCurrent": "true",
            "children": "Daily",
            "className": "nav-link",
            "name": "daily",
            "onClick": [Function],
            "to": "/daily",
          },
          "ref": null,
          "rendered": "Daily",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
